### ISSUE TITLE
[S3] Create Bucket fails when deploying to us-west-2

### ISSUE THREAD

**Issue by shogomuranushi** [k1LoW/certman#5]
```
$ certman request aaa.bbb.ccc.example.com
NOTICE! Your selected region is *us-west-2*. Certman create certificate on *us-west-2*. OK? Yes
NOTICE! When requesting, Certman replace Active Receipt Rule Set. OK? Yes
[✔] [ACM] Check Certificate (us-west-2) (successfull)
[✔] [Route53] Check Hosted Zone (us-west-2) (successfull)
[✔] [Route53] Check TXT Record (us-west-2) (successfull)
[✔] [Route53] Check MX Record (us-west-2) (successfull)
[|] [S3] Create Bucket for SES inbound (us-west-2)
Error: The specified bucket is not valid.
[✖] [S3] Create Bucket for SES inbound (us-west-2) (error)
Done.

certificate_arn:
```
Is there a way to output debug logs?

**Comment by k1LoW** [comment]
Hi @shogomuranushi . Thank you for your report.

> Error: The specified bucket is not valid.

S3 bucket name that certman generate with your request FQDN is invalid. 
( see https://github.com/k1LoW/certman/blob/master/lib/certman/client.rb#L174 )

I think that it was caught in limit of the number of bucket name characters. ( http://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html )

**Comment by k1LoW** [comment]
Fix S3 / SES Rule / Rule Set limitation support at v0.3.1. 
Could you try it ?

**Comment by shogomuranushi** [comment]
Thank you very much.
I tried 0.3.1. The results are as follows.

## us-east-1 => OK
```
$ AWS_DEFAULT_REGION=us-east-1
$ certman request aaaa.bbb.ccc.ddddd.ee
NOTICE! Your selected region is *us-east-1*. Certman create certificate on *us-east-1*. OK? Yes
NOTICE! When requesting, Certman replace Active Receipt Rule Set. OK? Yes
[✔] [ACM] Check Certificate (us-east-1) (successfull)
[✔] [Route53] Check Hosted Zone (us-east-1) (successfull)
[✔] [Route53] Check TXT Record (us-east-1) (successfull)
[✔] [Route53] Check MX Record (us-east-1) (successfull)
[✔] [S3] Create Bucket for SES inbound (us-east-1) (successfull)
[✔] [SES] Create Domain Identity (us-east-1) (successfull)
[✔] [Route53] Create TXT Record Set to verify Domain Identity (us-east-1) (successfull)
[✔] [SES] Check Domain Identity Status *verified* (us-east-1) (successfull)
[✔] [Route53] Create MX Record Set (us-east-1) (successfull)
[✔] [SES] Create Receipt Rule Set (us-east-1) (successfull)
[✔] [SES] Create Receipt Rule (us-east-1) (successfull)
[✔] [SES] Replace Active Receipt Rule Set (us-east-1) (successfull)
[✔] [ACM] Request Certificate (us-east-1) (successfull)
[✔] [S3] Check approval mail (will take about 30 min) (us-east-1) (successfull)
[✔] [SES] Revert Active Receipt Rule Set (us-east-1) (successfull)
[✔] [SES] Delete Receipt Rule (us-east-1) (successfull)
[✔] [SES] Delete Receipt Rule Set (us-east-1) (successfull)
[✔] [Route53] Delete MX Record Set (us-east-1) (successfull)
[✔] [Route53] Delete TXT Record Set (us-east-1) (successfull)
[✔] [SES] Delete Verified Domain Identiry (us-east-1) (successfull)
[✔] [S3] Delete Bucket (us-east-1) (successfull)
Done.

certificate_arn: arn:aws:acm:us-east-1:123456789012:certificate/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE
```

## ap-northeast-1 => OK
```
$ AWS_DEFAULT_REGION=ap-northeast-1
$ certman request aaaa.bbb.ccc.ddddd.ee
NOTICE! Your selected region is *ap-northeast-1*. Certman create certificate on *ap-northeast-1*. OK? (YNOTICE! Your selected region is *ap-northeast-1*. Certman create certificate on *ap-northeast-1*. OK? (YNOTICE! Your selected region is *ap-northeast-1*. Certman create certificate on *ap-northeast-1*. OK? Yes
NOTICE! Certman use *us-east-1* S3/SES. OK? Yes
NOTICE! When requesting, Certman replace Active Receipt Rule Set. OK? Yes
[✔] [ACM] Check Certificate (ap-northeast-1) (successfull)
[✔] [Route53] Check Hosted Zone (ap-northeast-1) (successfull)
[✔] [Route53] Check TXT Record (ap-northeast-1) (successfull)
[✔] [Route53] Check MX Record (ap-northeast-1) (successfull)
[✔] [S3] Create Bucket for SES inbound (us-east-1) (successfull)
[✔] [SES] Create Domain Identity (us-east-1) (successfull)
[✔] [Route53] Create TXT Record Set to verify Domain Identity (ap-northeast-1) (successfull)
[✔] [SES] Check Domain Identity Status *verified* (us-east-1) (successfull)
[✔] [Route53] Create MX Record Set (ap-northeast-1) (successfull)
[✔] [SES] Create Receipt Rule Set (us-east-1) (successfull)
[✔] [SES] Create Receipt Rule (us-east-1) (successfull)
[✔] [SES] Replace Active Receipt Rule Set (us-east-1) (successfull)
[✔] [ACM] Request Certificate (ap-northeast-1) (successfull)
[✔] [S3] Check approval mail (will take about 30 min) (us-east-1) (successfull)
[✔] [SES] Revert Active Receipt Rule Set (us-east-1) (successfull)
[✔] [SES] Delete Receipt Rule (us-east-1) (successfull)
[✔] [SES] Delete Receipt Rule Set (us-east-1) (successfull)
[✔] [Route53] Delete MX Record Set (ap-northeast-1) (successfull)
[✔] [Route53] Delete TXT Record Set (ap-northeast-1) (successfull)
[✔] [SES] Delete Verified Domain Identiry (us-east-1) (successfull)
[✔] [S3] Delete Bucket (us-east-1) (successfull)
Done.

certificate_arn: arn:aws:acm:ap-northeast-1:123456789012:certificate/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE
```

## us-west-2 => NG
```
$ AWS_DEFAULT_REGION=us-west-2
$ certman request aaaa.bbb.ccc.ddddd.ee
NOTICE! Your selected region is *us-west-2*. Certman create certificate on *us-west-2*. OK? Yes
NOTICE! When requesting, Certman replace Active Receipt Rule Set. OK? Yes
[✔] [ACM] Check Certificate (us-west-2) (successfull)
[✔] [Route53] Check Hosted Zone (us-west-2) (successfull)
[✔] [Route53] Check TXT Record (us-west-2) (successfull)
[✔] [Route53] Check MX Record (us-west-2) (successfull)
[\] [S3] Create Bucket for SES inbound (us-west-2)
Error: The specified bucket is not valid.
[✖] [S3] Create Bucket for SES inbound (us-west-2) (error)
Done.

certificate_arn:
```
I thought there was a problem when making S3 with us-west-2.
Therefore we deleted 'us-west-2' and 'eu-west-1' from 'lib/certman/cli.rb' and 'lib/certman/client.rb'.

```
- unless ['us-east-1', 'us-west-2', 'eu-west-1'].include?(Aws.config[:region])
+ unless ['us-east-1'].include?(Aws.config[:region])
```
Then I ran it without problems.
There seems to be a problem when creating S3 with us-west-2.
```
$ AWS_DEFAULT_REGION=us-west-2
$ certman request aaaa.bbb.ccc.ddddd.ee
NOTICE! Your selected region is *us-west-2*. Certman create certificate on *us-west-2*. OK? Yes
NOTICE! Certman use *us-east-1* S3/SES. OK? Yes
NOTICE! When requesting, Certman replace Active Receipt Rule Set. OK? Yes
[✔] [ACM] Check Certificate (us-west-2) (successfull)
[✔] [Route53] Check Hosted Zone (us-west-2) (successfull)
[✔] [Route53] Check TXT Record (us-west-2) (successfull)
[✔] [Route53] Check MX Record (us-west-2) (successfull)
[✔] [S3] Create Bucket for SES inbound (us-east-1) (successfull)
[✔] [SES] Create Domain Identity (us-east-1) (successfull)
[✔] [Route53] Create TXT Record Set to verify Domain Identity (us-west-2) (successfull)
[✔] [SES] Check Domain Identity Status *verified* (us-east-1) (successfull)
[✔] [Route53] Create MX Record Set (us-west-2) (successfull)
[✔] [SES] Create Receipt Rule Set (us-east-1) (successfull)
[✔] [SES] Create Receipt Rule (us-east-1) (successfull)
[✔] [SES] Replace Active Receipt Rule Set (us-east-1) (successfull)
[✔] [ACM] Request Certificate (us-west-2) (successfull)
[✔] [S3] Check approval mail (will take about 30 min) (us-east-1) (successfull)
[✔] [SES] Revert Active Receipt Rule Set (us-east-1) (successfull)
[✔] [SES] Delete Receipt Rule (us-east-1) (successfull)
[✔] [SES] Delete Receipt Rule Set (us-east-1) (successfull)
[✔] [Route53] Delete MX Record Set (us-west-2) (successfull)
[✔] [Route53] Delete TXT Record Set (us-west-2) (successfull)
[✔] [SES] Delete Verified Domain Identiry (us-east-1) (successfull)
[✔] [S3] Delete Bucket (us-east-1) (successfull)
Done.

certificate_arn: arn:aws:acm:us-west-2:123456789012:certificate/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE
```
S3 and SES are temporarily used only.
How about making S3 and SES with us-east-1?

**Comment by k1LoW** [comment]
Hi @shogomuranushi . Thank you for your report!
I'll check it.

**Comment by k1LoW** [comment]
Hi @shogomuranushi .
Fix S3 invalid bucket name at v0.4.0.
Could you try it ?

**Comment by shogomuranushi** [comment]
Sorry.
`us-west-2` fails with` Check approval mail`
```
certman request aaaaaa.bbb.ccc.ddddd.ee
NOTICE! Your selected region is *us-west-2*. Certman create certificate on *us-west-2*. OK? Yes
NOTICE! When requesting, Certman replace Active Receipt Rule Set. OK? Yes
[✔] [ACM] Check Certificate (us-west-2) (successfull)
[✔] [Route53] Check Hosted Zone (us-west-2) (successfull)
[✔] [Route53] Check TXT Record (us-west-2) (successfull)
[✔] [Route53] Check MX Record (us-west-2) (successfull)
[✔] [S3] Create Bucket for SES inbound (us-west-2) (successfull)
[✔] [SES] Create Domain Identity (us-west-2) (successfull)
[✔] [Route53] Create TXT Record Set to verify Domain Identity (us-west-2) (successfull)
[✔] [SES] Check Domain Identity Status *verified* (us-west-2) (successfull)
[✔] [Route53] Create MX Record Set (us-west-2) (successfull)
[✔] [SES] Create Receipt Rule Set (us-west-2) (successfull)
[✔] [SES] Create Receipt Rule (us-west-2) (successfull)
[✔] [SES] Replace Active Receipt Rule Set (us-west-2) (successfull)
[✔] [ACM] Request Certificate (us-west-2) (successfull)
[/] [S3] Check approval mail (will take about 30 min) (us-west-2)
Error: Can not approve
[✖] [S3] Check approval mail (will take about 30 min) (us-west-2) (error)
[✔] [SES] Revert Active Receipt Rule Set (us-west-2) (successfull)
[✔] [SES] Delete Receipt Rule (us-west-2) (successfull)
[✔] [SES] Delete Receipt Rule Set (us-west-2) (successfull)
[✔] [Route53] Delete MX Record Set (us-west-2) (successfull)
[✔] [Route53] Delete TXT Record Set (us-west-2) (successfull)
[✔] [SES] Delete Verified Domain Identiry (us-west-2) (successfull)
[✔] [S3] Delete Bucket (us-west-2) (successfull)
Done.

certificate_arn: arn:aws:acm:us-west-2:123456789012:certificate/aaaaaaaaaaaaaaaaaaaaaaaaaaa
```

**Comment by k1LoW** [comment]
@shogomuranushi .

> us-west-2 fails withCheck approval mail

This bug fixed by #16 . Sorry! 🙇 🙇 🙇 🙇 🙇

**Comment by shogomuranushi** [comment]
It's ok!!
you are perfect human!

```
$ certman request aaaaaa.bbb.ccc.ddddd.ee
NOTICE! Your selected region is *us-west-2*. Certman create certificate on *us-west-2*. OK? (Y/NOTICE! Your selected region is *us-west-2*. Certman create certificate on *us-west-2*. OK? (Y/NOTICE! Your selected region is *us-west-2*. Certman create certificate on *us-west-2*. OK? Yes
NOTICE! When requesting, Certman replace Active Receipt Rule Set. OK? Yes
[✔] [ACM] Check Certificate (us-west-2) (successfull)
[✔] [Route53] Check Hosted Zone (us-west-2) (successfull)
[✔] [Route53] Check TXT Record (us-west-2) (successfull)
[✔] [Route53] Check MX Record (us-west-2) (successfull)
[✔] [S3] Create Bucket for SES inbound (us-west-2) (successfull)
[✔] [SES] Create Domain Identity (us-west-2) (successfull)
[✔] [Route53] Create TXT Record Set to verify Domain Identity (us-west-2) (successfull)
[✔] [SES] Check Domain Identity Status *verified* (us-west-2) (successfull)
[✔] [Route53] Create MX Record Set (us-west-2) (successfull)
[✔] [SES] Create Receipt Rule Set (us-west-2) (successfull)
[✔] [SES] Create Receipt Rule (us-west-2) (successfull)
[✔] [SES] Replace Active Receipt Rule Set (us-west-2) (successfull)
[✔] [ACM] Request Certificate (us-west-2) (successfull)
[✔] [S3] Check approval mail (will take about 30 min) (us-west-2) (successfull)
[✔] [SES] Revert Active Receipt Rule Set (us-west-2) (successfull)
[✔] [SES] Delete Receipt Rule (us-west-2) (successfull)
[✔] [SES] Delete Receipt Rule Set (us-west-2) (successfull)
[✔] [Route53] Delete MX Record Set (us-west-2) (successfull)
[✔] [Route53] Delete TXT Record Set (us-west-2) (successfull)
[✔] [SES] Delete Verified Domain Identiry (us-west-2) (successfull)
[✔] [S3] Delete Bucket (us-west-2) (successfull)
Done.

certificate_arn: arn:aws:acm:us-west-2:123456789012:certificate/aaaaaaaaaaaaaaaaaaaaaaaaaaa
```
